builtin/logical/database/path_roles.go

```diff
@@ -839,6 +839,15 @@ func (s *staticAccount) NextRotationTime() time.Time {
 	return s.Schedule.Next(s.LastVaultRotation)
 }
 
+// NextRotationTimeFromLast calculates the next rotation time for period and
+// schedule-based roles based on the last rotation time.
+func (s *staticAccount) NextRotationTimeFromLast(last time.Time) time.Time {
```

Comment:
nit: maybe we should change this to `NextRotationTimeFromInput(input time.Time)`? The main reason is that I think it helps to differentiate it from the `NextRotationTime()` method, albeit subtly. `NextRotationTime()` evaluates its return based on the method receiver's LastVaultRotation, whereas this method evaluates it from a given input which may not actually be the last vault rotation. Although, that is the only current use case we have for it, it doesn't prescribe it.

Reply:
As a side note, I think if we did some refactoring we wouldn't even need this method and could use the above `NextRotationTime()` in the manual rotate path.